File: src/test/java/com/baulsupp/oksocial/integration/TestCredentialsStore.java
package com.baulsupp.oksocial.integration;

import com.baulsupp.oksocial.credentials.CredentialsStore;
import com.baulsupp.oksocial.credentials.ServiceDefinition;
import com.google.common.collect.Maps;
import java.util.Map;
import java.util.Optional;

import static java.util.Optional.ofNullable;

public class TestCredentialsStore implements CredentialsStore {
  public Map<String, String> tokens = Maps.newHashMap();

  @Override public <T> Optional<T> readDefaultCredentials(ServiceDefinition<T> serviceDefinition) {
    return ofNullable(tokens.get(serviceDefinition.apiHost()))
        .map(serviceDefinition::parseCredentialsString);
  }

  @Override
  public <T> void storeCredentials(T credentials, ServiceDefinition<T> serviceDefinition) {
    tokens.put(serviceDefinition.apiHost(), serviceDefinition.formatCredentialsString(credentials));
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/i9n/TestCredentialsStore.kt
package com.baulsupp.oksocial.i9n

import com.baulsupp.oksocial.credentials.CredentialsStore
import com.baulsupp.oksocial.credentials.ServiceDefinition

// TODO use token set
class TestCredentialsStore : CredentialsStore {
  var tokens: MutableMap<String, String> = linkedMapOf()

  override fun <T> get(serviceDefinition: ServiceDefinition<T>, tokenSet: String?): T? {
    return tokens[serviceDefinition.apiHost()]
        ?.let { serviceDefinition.parseCredentialsString(it) }
  }

  override fun <T> set(
          serviceDefinition: ServiceDefinition<T>, tokenSet: String?, credentials: T) {
    tokens.put(serviceDefinition.apiHost(), serviceDefinition.formatCredentialsString(credentials))
  }

  override fun <T> remove(serviceDefinition: ServiceDefinition<T>, tokenSet: String?) {
    tokens.remove(serviceDefinition.apiHost())
  }
}
